### PR TITLE
[RSN-23] - Create constraints

### PR DESCRIPTION
**Pull request by rogacky11**
## Description
I have created constraints for individual rows which are checking if added value's format is correct.

## Related issue and/or ticket
[RSN-23](https://toopricey.youtrack.cloud/issue/RSN-23/DB-Create-constraints)

### CODE REVIEW

File: Database/init-db.sql [hunk 1]
```diff
@@ -36,22 +36,23 @@ CREATE TABLE IF NOT EXISTS events.tag (
...
+  "tag_id" integer NOT NULL,
+  PRIMARY KEY (event_id, tag_id) 
 );
  
 CREATE TABLE IF NOT EXISTS users.user (
   "id" SERIAL PRIMARY KEY,
   "name" varchar(255) NOT NULL,
-  "surname" varchar(255) NOT NULL,
+  "surname" varchar(255) NOT NULL UNIQUE,
   "username" varchar(255) NOT NULL,
```
[the remark this answers is not in the document]
Comment (rogacky11) [reply Database/init-db.sql]: It was my oversight. Already changed.

File: Database/init-db.sql [hunk 2]
```diff
@@ -101,13 +103,14 @@ CREATE TABLE IF NOT EXISTS common.object_type (
...
+  "interest_id" integer NOT NULL,
+  "level" integer NOT NULL,
+  PRIMARY KEY (user_id, interest_id)
 );
  
 CREATE TABLE IF NOT EXISTS users.interest (
   "id" SERIAL PRIMARY KEY,
-  "name" varchar(255) NOT NULL,
-  "level" integer NOT NULL
+  "name" varchar(255) NOT NULL
```
[the remark this answers is not in the document]
Comment (rogacky11) [reply Database/init-db.sql]: Same as examples from your comment above, added more unique values.

**Comment by rogacky11** [comment]
> To check:
> 
>     * events.participant: combination of event_id and user_id should be unique, we can even do it as primary key I think
> 
>     * common.status: status name should be unique
> 
>     * events.tag: tag name should be unique
> 
>     * events.parameter: combination of key, value should be unique I think (to discuss)
> 
>     * users.role: role name should be unique
> 
>     * common.object_type: object type name should be unique
> 
>     * common.image: we should check that object of type user have max of 1 image
> 
> 
> Also please merge main into your branch before merging, it will be easier to resolve conflicts on your computer than on github just before merging.

Every unique value mentioned here has been added. Furthermore to resolve problem with common.image and adding max 1 image for user, I have created unique index for object_type_id and object_id connection when object_type_id = 2 (it's user's obj_type_id)

**Review by raczu — APPROVED** [review]
lgtm

**Review by wzarek — APPROVED** [review]
OK

**Comment by wzarek** [comment]
@rogacky11 please rebase before merging and do squash and merge then.

**Pull request merged**